core/system/src/mpu/vmpu_freescale_k64.c

```diff
@@ -296,3 +295,36 @@ void vmpu_arch_init(void)
     /* init memory protection */
     vmpu_mem_init();
 }
+
+int vmpu_is_region_size_valid(uint32_t size)
+{
+    /* Align size to 32B */
+    const uint32_t masked_size = size & ~31;
+    if (masked_size < 32 || (1 << 29) < masked_size) {
+        /* 2^5 == 32, which is the minimum region size */
+        /* 2^29 == 512M, which is the maximum region size */
+        return 0;
+    }
+    /* There is no rounding, we only care about an exact match! */
+    return (masked_size == size);
+}
+
+uint32_t vmpu_round_up_region(uint32_t addr, uint32_t size)
```

Comment:
Can the implementation of `vmpu_round_up_region` be shared between these two MPUs? The code looks pretty much the same, other than which `vmpu_region_size_valid` function to call.